engine/framework/model/typedefs.go

```diff
@@ -27,3 +33,78 @@ type (
 	// Epoch is an increasing only value.
 	Epoch = int64
 )
+
+// Defines all task type
+// TODO: Refine me.Currently, when adding a new worker type or job type, we need to modify many code places,
+// NOTICE: DO NOT CHANGE the previous worker type
+// Modify the comment in model IF you add some new worker type
+const (
+	JobManager = WorkerType(iota + 1)
+	// job master
+	CvsJobMaster
+	FakeJobMaster
+	DMJobMaster
+	CdcJobMaster
+	// task
+	CvsTask
+	FakeTask
+	DmTask
+	CdcTask
+	// worker
+	WorkerDMDump
+	WorkerDMLoad
+	WorkerDMSync
+	// extend the worker type here
+)
+
+var typesStringify = [...]string{
+	0:             "",
+	JobManager:    "JobManager",
+	CvsJobMaster:  "CVSJobMaster",
+	FakeJobMaster: "FakeJobMaster",
+	DMJobMaster:   "DMJobMaster",
+	CdcJobMaster:  "CDCJobMaster",
+	CvsTask:       "CVSTask",
+	FakeTask:      "FakeTask",
+	DmTask:        "DMTask",
+	CdcTask:       "CDCTask",
+	WorkerDMDump:  "DMDumpTask",
+	WorkerDMLoad:  "DMLoadTask",
+	WorkerDMSync:  "DMSyncTask",
+}
+
+var toWorkerType map[string]WorkerType
+
+func init() {
+	toWorkerType = make(map[string]WorkerType, len(typesStringify))
+	for i, s := range typesStringify {
+		toWorkerType[s] = WorkerType(i)
+	}
+}
+
+// String implements fmt.Stringer interface
+func (wt WorkerType) String() string {
+	if int(wt) >= len(typesStringify) {
+		return fmt.Sprintf("Unknown WorkerType %d", wt)
+	}
+	return typesStringify[wt]
+}
+
+// MarshalJSON marshals the enum as a quoted json string
+func (wt WorkerType) MarshalJSON() ([]byte, error) {
+	buffer := bytes.NewBufferString(`"`)
+	buffer.WriteString(wt.String())
+	buffer.WriteString(`"`)
+	return buffer.Bytes(), nil
+}
+
+// UnmarshalJSON unmashals a quoted json string to the enum value
+func (wt *WorkerType) UnmarshalJSON(b []byte) error {
+	var j string
+	err := json.Unmarshal(b, &j)
+	if err != nil {
+		return err
+	}
+	*wt = toWorkerType[j]
```

Comment:
it's easier to locate the error by returning error when the string is not in toWorkerType